pkg/git/v2/transport.go

```diff
@@ -57,26 +61,49 @@ func (s *BasicAuth) Method(secret corev1.Secret) (*common.Auth, error) {
 	if d, ok := secret.Data["password"]; ok {
 		password = string(d)
 	}
-	if username == "" || password == "" {
-		return nil, fmt.Errorf("invalid '%s' secret data: required fields 'username' and 'password'", secret.Name)
+	if username != "" && password != "" {
```

Comment:
This behavior change is needed as it should be possible to have cert validation on repositories with no authentication. This will change libgit2 behavior to not throw an error if username and password are missing if a secret is specified.